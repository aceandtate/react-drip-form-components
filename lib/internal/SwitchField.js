'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _templateObject = _taggedTemplateLiteral(['\n  margin: 0.4em 0;\n  color: ', ';\n'], ['\n  margin: 0.4em 0;\n  color: ', ';\n']);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _Field = require('./Field');

var _Field2 = _interopRequireDefault(_Field);

var _theme = require('../theme');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var SwitchField = (0, _styledComponents2.default)(_Field2.default)(_templateObject, function (props) {
  return props.disabled ? (0, _theme.get)(props, 'disabled') : (0, _theme.get)(props, 'color');
});

var _default = SwitchField;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(SwitchField, 'SwitchField', 'src/internal/SwitchField.js');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/internal/SwitchField.js');
}();

;