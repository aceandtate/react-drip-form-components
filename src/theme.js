export const colors = {
  white: 'rgb(255, 255, 255)',
  grayDark: 'rgb(54, 54, 54)',
  gray: 'rgb(204, 204, 204)',
  grayLight: 'rgb(187, 187, 187)',
  grayLightest: 'rgb(219, 219, 219)',
  primary: 'rgb(0, 188, 212)',
  error: 'rgb(245, 0, 87)',
  disabled: 'rgb(245, 245, 245)',
  shadow: 'rgba(0, 0, 0, 0.04)',
};

export const theme = {
  // typography
  fontSize: '1rem',
  color: colors.grayDark,
  error: colors.error,
  disabled: colors.gray,

  // background
  background: colors.white,
  backgroundEnter: colors.white,
  backgroundFocus: colors.white,
  backgroundDisabled: colors.disabled,
  backgroundError: colors.white,

  // border
  borderWidth: '1px',
  borderRadius: '3px',
  borderColor: colors.grayLightest,
  borderColorEnter: colors.gray,
  borderColorFocus: colors.primary,
  borderColorDisabled: colors.disabled,
  borderColorError: colors.error,

  // box-shadow
  boxShadow: '0 0 0 transparent',
  boxShadowEnter: `0 2px 2px ${colors.shadow} inset`,
  boxShadowDisabled: `0 2px 2px ${colors.shadow} inset`,

  // transition
  transitionProperty: 'color, box-shadow, border',
  transitionDuration: '110ms',
  transitionTimingFunction: 'cubic-bezier(.25, .46, .45, .94)', // ease-out-quad

  // control
  controlHeight: '2.7em',

  // select
  selectArrowColor: colors.primary,

  // spinner
  spinnerColor: colors.primary,
  spinnerDuration: '500ms',
};

export const extendTheme = (extendsTheme = {}) => ({
  ...theme,
  ...extendsTheme,
});

export const themeProps = {
  theme: {
    rdf: theme,
  },
};

export const get = (props, key) => props.theme.rdf[key];

export const bind = key => props => get(props, key);
