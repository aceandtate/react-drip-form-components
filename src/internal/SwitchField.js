import styled from 'styled-components';

import Field from './Field';
import { get } from '../theme';

const SwitchField = styled(Field)`
  margin: 0.4em 0;
  color: ${props => props.disabled
    ? get(props, 'disabled')
    : get(props, 'color')};
`;

export default SwitchField;
