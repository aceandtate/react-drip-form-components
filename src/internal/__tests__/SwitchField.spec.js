import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import { theme } from '../../theme';
import SwitchField from '../SwitchField';
import { mountWithTheme } from '../../__tests__/utils';


describe('Internal#<SwitchField />', () => {
  test('Should be render children', () => {
    const element = <div id="foo">foo bar baz</div>;
    const wrapper = shallow(
      <SwitchField>
        {element}
      </SwitchField>
    );

    expect(wrapper.contains(element)).toBe(true);
  });


  test('Should be styled component', () => {
    const wrapper = shallow(<SwitchField />);
    expect(wrapper).toHaveStyleRule('color', theme.color);

    wrapper.setProps({ disabled: true });
    expect(wrapper).toHaveStyleRule('color', theme.disabled);
  });


  test('Should be changed styled by theme', () => {
    const testTheme = {
      color: 'hotpink',
      disabled: 'green',
    };

    let wrapper;

    wrapper = mountWithTheme(<SwitchField />, testTheme);
    expect(wrapper).toHaveStyleRule('color', 'hotpink');

    wrapper = mountWithTheme(<SwitchField disabled />, testTheme);
    expect(wrapper).toHaveStyleRule('color', 'green');
  });
});
