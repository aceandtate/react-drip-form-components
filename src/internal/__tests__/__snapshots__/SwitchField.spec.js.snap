// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Internal#<SwitchField /> Should be changed styled by theme 1`] = `
.c0 {
  display: block;
  position: relative;
  margin: 0;
  font-size: 1rem;
  margin: 0.4em 0;
  color: hotpink;
}

<ThemeProvider
  theme={
    Object {
      "rdf": Object {
        "background": "rgb(255, 255, 255)",
        "backgroundDisabled": "rgb(245, 245, 245)",
        "backgroundEnter": "rgb(255, 255, 255)",
        "backgroundError": "rgb(255, 255, 255)",
        "backgroundFocus": "rgb(255, 255, 255)",
        "borderColor": "rgb(219, 219, 219)",
        "borderColorDisabled": "rgb(245, 245, 245)",
        "borderColorEnter": "rgb(204, 204, 204)",
        "borderColorError": "rgb(245, 0, 87)",
        "borderColorFocus": "rgb(0, 188, 212)",
        "borderRadius": "3px",
        "borderWidth": "1px",
        "boxShadow": "0 0 0 transparent",
        "boxShadowDisabled": "0 2px 2px rgba(0, 0, 0, 0.04) inset",
        "boxShadowEnter": "0 2px 2px rgba(0, 0, 0, 0.04) inset",
        "color": "hotpink",
        "controlHeight": "2.7em",
        "disabled": "green",
        "error": "rgb(245, 0, 87)",
        "fontSize": "1rem",
        "selectArrowColor": "rgb(0, 188, 212)",
        "spinnerColor": "rgb(0, 188, 212)",
        "spinnerDuration": "500ms",
        "transitionDuration": "110ms",
        "transitionProperty": "color, box-shadow, border",
        "transitionTimingFunction": "cubic-bezier(.25, .46, .45, .94)",
      },
    }
  }
>
  <styled.div
    theme={
      Object {
        "rdf": Object {
          "background": "rgb(255, 255, 255)",
          "backgroundDisabled": "rgb(245, 245, 245)",
          "backgroundEnter": "rgb(255, 255, 255)",
          "backgroundError": "rgb(255, 255, 255)",
          "backgroundFocus": "rgb(255, 255, 255)",
          "borderColor": "rgb(219, 219, 219)",
          "borderColorDisabled": "rgb(245, 245, 245)",
          "borderColorEnter": "rgb(204, 204, 204)",
          "borderColorError": "rgb(245, 0, 87)",
          "borderColorFocus": "rgb(0, 188, 212)",
          "borderRadius": "3px",
          "borderWidth": "1px",
          "boxShadow": "0 0 0 transparent",
          "boxShadowDisabled": "0 2px 2px rgba(0, 0, 0, 0.04) inset",
          "boxShadowEnter": "0 2px 2px rgba(0, 0, 0, 0.04) inset",
          "color": "rgb(54, 54, 54)",
          "controlHeight": "2.7em",
          "disabled": "rgb(204, 204, 204)",
          "error": "rgb(245, 0, 87)",
          "fontSize": "1rem",
          "selectArrowColor": "rgb(0, 188, 212)",
          "spinnerColor": "rgb(0, 188, 212)",
          "spinnerDuration": "500ms",
          "transitionDuration": "110ms",
          "transitionProperty": "color, box-shadow, border",
          "transitionTimingFunction": "cubic-bezier(.25, .46, .45, .94)",
        },
      }
    }
  >
    <div
      className="c0"
    />
  </styled.div>
</ThemeProvider>
`;

exports[`Internal#<SwitchField /> Should be changed styled by theme 2`] = `
.c0 {
  display: block;
  position: relative;
  margin: 0;
  font-size: 1rem;
  margin: 0.4em 0;
  color: green;
}

<ThemeProvider
  theme={
    Object {
      "rdf": Object {
        "background": "rgb(255, 255, 255)",
        "backgroundDisabled": "rgb(245, 245, 245)",
        "backgroundEnter": "rgb(255, 255, 255)",
        "backgroundError": "rgb(255, 255, 255)",
        "backgroundFocus": "rgb(255, 255, 255)",
        "borderColor": "rgb(219, 219, 219)",
        "borderColorDisabled": "rgb(245, 245, 245)",
        "borderColorEnter": "rgb(204, 204, 204)",
        "borderColorError": "rgb(245, 0, 87)",
        "borderColorFocus": "rgb(0, 188, 212)",
        "borderRadius": "3px",
        "borderWidth": "1px",
        "boxShadow": "0 0 0 transparent",
        "boxShadowDisabled": "0 2px 2px rgba(0, 0, 0, 0.04) inset",
        "boxShadowEnter": "0 2px 2px rgba(0, 0, 0, 0.04) inset",
        "color": "hotpink",
        "controlHeight": "2.7em",
        "disabled": "green",
        "error": "rgb(245, 0, 87)",
        "fontSize": "1rem",
        "selectArrowColor": "rgb(0, 188, 212)",
        "spinnerColor": "rgb(0, 188, 212)",
        "spinnerDuration": "500ms",
        "transitionDuration": "110ms",
        "transitionProperty": "color, box-shadow, border",
        "transitionTimingFunction": "cubic-bezier(.25, .46, .45, .94)",
      },
    }
  }
>
  <styled.div
    disabled={true}
    theme={
      Object {
        "rdf": Object {
          "background": "rgb(255, 255, 255)",
          "backgroundDisabled": "rgb(245, 245, 245)",
          "backgroundEnter": "rgb(255, 255, 255)",
          "backgroundError": "rgb(255, 255, 255)",
          "backgroundFocus": "rgb(255, 255, 255)",
          "borderColor": "rgb(219, 219, 219)",
          "borderColorDisabled": "rgb(245, 245, 245)",
          "borderColorEnter": "rgb(204, 204, 204)",
          "borderColorError": "rgb(245, 0, 87)",
          "borderColorFocus": "rgb(0, 188, 212)",
          "borderRadius": "3px",
          "borderWidth": "1px",
          "boxShadow": "0 0 0 transparent",
          "boxShadowDisabled": "0 2px 2px rgba(0, 0, 0, 0.04) inset",
          "boxShadowEnter": "0 2px 2px rgba(0, 0, 0, 0.04) inset",
          "color": "rgb(54, 54, 54)",
          "controlHeight": "2.7em",
          "disabled": "rgb(204, 204, 204)",
          "error": "rgb(245, 0, 87)",
          "fontSize": "1rem",
          "selectArrowColor": "rgb(0, 188, 212)",
          "spinnerColor": "rgb(0, 188, 212)",
          "spinnerDuration": "500ms",
          "transitionDuration": "110ms",
          "transitionProperty": "color, box-shadow, border",
          "transitionTimingFunction": "cubic-bezier(.25, .46, .45, .94)",
        },
      }
    }
  >
    <div
      className="c0"
      disabled={true}
    />
  </styled.div>
</ThemeProvider>
`;

exports[`Internal#<SwitchField /> Should be styled component 1`] = `
.c0 {
  display: block;
  position: relative;
  margin: 0;
  font-size: 1rem;
  margin: 0.4em 0;
  color: rgb(54,54,54);
}

<div
  className="c0"
/>
`;

exports[`Internal#<SwitchField /> Should be styled component 2`] = `
.c0 {
  display: block;
  position: relative;
  margin: 0;
  font-size: 1rem;
  margin: 0.4em 0;
  color: rgb(204,204,204);
}

<div
  className="c0"
  disabled={true}
/>
`;
